src/main/java/com/hivemq/mqtt/topic/SubscriberWithIdentifiers.java

```diff
@@ -52,11 +52,12 @@ public SubscriberWithIdentifiers(final @NotNull String subscriber, final int qos
     public SubscriberWithIdentifiers(final @NotNull String subscriber, final int qos, final byte flags, final @Nullable String sharedName,
                                      final @NotNull ImmutableList<Integer> subscriptionIdentifier, final @Nullable String topicFilter) {
         checkNotNull(subscriber, "Subscriber must not be null");
+        checkNotNull(subscriptionIdentifier, "Subscriber Identifiers must not be null")
         this.subscriber = subscriber;
         this.qos = qos;
         this.flags = flags;
         this.sharedName = sharedName;
-        this.subscriptionIdentifier = ImmutableIntArray.of();
+        this.subscriptionIdentifier = (subscriptionIdentifier == null) ? ImmutableIntArray.of():ImmutableIntArray.copyOf(subscriptionIdentifier);
```

Comment:
With the added `checkNotNull` the null check here is, as you already mentioned, not necessary anymore and can be removed.